tests/storage/test_background_update.py

```diff
@@ -116,14 +115,15 @@ def prepare(self, reactor, clock, homeserver):
         )
 
         # Mock out the AsyncContextManager
-        self._update_ctx_manager = Mock(spec=["__aenter__", "__aexit__"])
-        self._update_ctx_manager.__aenter__ = Mock(
-            return_value=make_awaitable(None),
-        )
-        self._update_ctx_manager.__aexit__ = Mock(return_value=make_awaitable(None))
+        class MockCM:
+            pass
+
+        self._update_ctx_manager = MockCM
+        self._update_ctx_manager.__aenter__ = simple_async_mock(return_value=None)
+        self._update_ctx_manager.__aexit__ = simple_async_mock(return_value=None)
```

Comment:
Would it work to define these on the class directly? That seems a bit cleaner!

Reply:
Yes!